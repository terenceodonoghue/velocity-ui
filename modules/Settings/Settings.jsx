/** @jsx jsx */
import { jsx } from '@emotion/core';
import { Button, Container, Input } from 'components';
import * as S from './Settings.styles';

const Personal = () => (
  <S.Personal>
    <Container.Card heading="Personal data">
      <S.Content>
        <S.Paragraph>
          Use this page to update your contact information and change your
          password.
        </S.Paragraph>
        <S.Options>
          <S.Option>
            <Input.Label>
              Email address
              <Input.Text type="text" defaultValue="appleseed_jane@mac.com" />
            </Input.Label>
          </S.Option>
        </S.Options>
        <S.Options>
          <S.Option>
            <Input.Label>
              First name
              <Input.Text type="text" defaultValue="Jane" />
            </Input.Label>
          </S.Option>
          <S.Option>
            <Input.Label>
              Last name
              <Input.Text type="text" defaultValue="Appleseed" />
            </Input.Label>
          </S.Option>
          <S.Option>
            <Input.Label>
              Birth date
              <Input.Text type="date" defaultValue="1965-05-25" />
            </Input.Label>
          </S.Option>
        </S.Options>
        <S.Options>
          <S.Option>
            <Input.Label>
              Current password
              <Input.Text type="password" defaultValue="password" />
            </Input.Label>
          </S.Option>
          <S.Option>
            <Input.Label>
              New password
              <Input.Text type="password" />
            </Input.Label>
          </S.Option>
          <S.Option>
            <Input.Label>
              Confirm
              <Input.Text type="password" />
            </Input.Label>
          </S.Option>
        </S.Options>
        <S.Options>
          <S.Option>
            <Button.Contained>Save Changes</Button.Contained>
          </S.Option>
        </S.Options>
      </S.Content>
    </Container.Card>
  </S.Personal>
);

const Integrations = () => (
  <S.Integrations>
    <Container.Card heading="Integrations">
      <S.Content>
        <S.Paragraph>Manage third-party app integrations.</S.Paragraph>
      </S.Content>
    </Container.Card>
  </S.Integrations>
);

const Notifications = () => (
  <S.Notifications>
    <Container.Card heading="Notifications">
      <S.Content>
        <S.Paragraph>
          Control your notification and auto-follow settings.
        </S.Paragraph>
        <S.Options>
          <S.Option>
            <S.Text>
              <S.Label>Email Notification</S.Label>
              <S.Description>Commits data and history</S.Description>
            </S.Text>
            <Input.Switch />
          </S.Option>
          <S.Option>
            <S.Text>
              <S.Label>Monthly Reports</S.Label>
              <S.Description>Commits data and history</S.Description>
            </S.Text>
            <Input.Switch />
          </S.Option>
          <S.Option>
            <S.Text>
              <S.Label>Push Notification</S.Label>
              <S.Description>Commits data and history</S.Description>
            </S.Text>
            <Input.Switch defaultChecked />
          </S.Option>
          <S.Option>
            <S.Text>
              <S.Label>Quarter Reports</S.Label>
              <S.Description>Commits data and history</S.Description>
            </S.Text>
            <Input.Switch />
          </S.Option>
        </S.Options>
      </S.Content>
    </Container.Card>
  </S.Notifications>
);

export default {
  Personal,
  Integrations,
  Notifications,
};
