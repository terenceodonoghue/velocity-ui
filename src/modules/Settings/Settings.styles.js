import styled from '@emotion/styled';
import { Buttons, Inputs } from '../../components';

export const CheckboxField = styled.label`
  flex: 0 1 365px;
  padding: 13px 11px;
`;

export const CheckboxText = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  border: ${(props) =>
    `solid 1px ${props.active ? props.theme.colors.blue : '#e0e7ff'}`};
  border-radius: 5px;
  padding: 19px 21px;
  ${(props) =>
    props.active &&
    `&::after {
    position: absolute;
    right: -8px;
    top: -8px;
    content: '';
    background-color: ${props.theme.colors.blue};
    background-image: url(./check-white.svg);
    background-position: center;
    background-repeat: no-repeat;
    border-radius: 50%;
    height: 18px;
    width: 18px;
  }`}
`;

export const Description = styled.p`
  margin: 0;
  color: ${(props) => props.theme.colors.blueyGrey};
  font-weight: ${(props) => props.theme.typography.fontWeightRegular};
  line-height: 1.47;
`;

export const FieldSet = styled.div`
  display: flex;
  flex-flow: row wrap;
  margin: 0 -11px;
  padding: 0 8px;
`;

export const Icon = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 50%;
  height: 70px;
  margin: 0 12px 0 0;
  width: 70px;
  background-color: rgba(128, 151, 177, 0.1);
`;

export const Label = styled.p`
  margin: 0;
  font-weight: ${(props) => props.theme.typography.fontWeightMedium};
  line-height: 1.47;
`;

export const Legend = styled.p`
  flex-basis: 100%;
  margin: 0 0 25px 0;
  padding: 0 12px;
  line-height: 1.47;
`;

export const Palette = styled.div`
  border-radius: 4px;
  height: 65px;
  width: 120px;
  background: ${(props) =>
    `linear-gradient(90deg, ${props.colors[0]} 33.33%, ${props.colors[1]} 33.33%, ${props.colors[1]} 66.66%, ${props.colors[2]} 66.66%) no-repeat;`};
`;

export const RadioField = styled(Inputs.Radio)`
  padding: 12px 0;
`;

export const SaveButton = styled(Buttons.Contained)`
  margin: 13px 0 18px 8px;
`;

export const SwitchField = styled(Inputs.Switch)`
  margin: 0 87px 0 0;
  padding: 10px 11px;
`;

export const TextField = styled(Inputs.Text)`
  flex: 0 1 290px;
  padding: 12px 11px;

  &:first-of-type {
    flex: 0 1 100%;
  }
`;

export const Theme = styled.div`
  padding: 8px 11px;
  margin: 0 13px 0 0;
`;
