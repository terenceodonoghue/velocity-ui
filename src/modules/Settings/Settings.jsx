/** @jsx jsx */
import { jsx } from '@emotion/core';
import { Container } from 'components';
import * as S from './Settings.styles';

const Personal = () => (
  <Container.Card heading="Personal data">
    <S.FieldSet>
      <S.Legend>
        Use this page to update your contact information and change your
        password.
      </S.Legend>
      <S.TextField defaultValue="appleseed_jane@mac.com" label="Email address" type="text" />
      <S.TextField defaultValue="Jane" label="First name" type="text" />
      <S.TextField defaultValue="Appleseed" label="Last name" type="text" />
      <S.TextField defaultValue="1965-05-25" label="Birth date" type="date" />
      <S.TextField defaultValue="password" label="Current password" type="password" />
      <S.TextField label="New password" type="password" />
      <S.TextField label="Confirm" type="password" />
    </S.FieldSet>
    <S.SaveButton>Save Changes</S.SaveButton>
  </Container.Card>
);

const Integrations = () => (
  <Container.Card heading="Integrations">
    <S.FieldSet>
      <S.Legend>Manage third-party app integrations.</S.Legend>
      <S.CheckboxField>
        <S.CheckboxText active>
          <S.Icon>
            <img alt="InVision" src="./invision.svg" />
          </S.Icon>
          <div>
            <S.Label>InVision</S.Label>
            <S.Description>Boards and prototypes</S.Description>
          </div>
        </S.CheckboxText>
      </S.CheckboxField>
      <S.CheckboxField>
        <S.CheckboxText active>
          <S.Icon>
            <img alt="GitHub" src="./github.svg" />
          </S.Icon>
          <div>
            <S.Label>GitHub</S.Label>
            <S.Description>Commits data and history</S.Description>
          </div>
        </S.CheckboxText>
      </S.CheckboxField>
      <S.CheckboxField>
        <S.CheckboxText>
          <S.Icon>
            <img alt="Slack" src="./slack.svg" />
          </S.Icon>
          <div>
            <S.Label>Slack</S.Label>
            <S.Description>Messages and channels</S.Description>
          </div>
        </S.CheckboxText>
      </S.CheckboxField>
      <S.CheckboxField>
        <S.CheckboxText>
          <S.Icon>
            <img alt="Mailchimp" src="./mailchimp.svg" />
          </S.Icon>
          <div>
            <S.Label>Mailchimp</S.Label>
            <S.Description>Subscribers list</S.Description>
          </div>
        </S.CheckboxText>
      </S.CheckboxField>
      <S.CheckboxField>
        <S.CheckboxText>
          <S.Icon>
            <img alt="Twitter" src="./twitter.svg" />
          </S.Icon>
          <div>
            <S.Label>Twitter</S.Label>
            <S.Description>Tweets data</S.Description>
          </div>
        </S.CheckboxText>
      </S.CheckboxField>
      <S.CheckboxField>
        <S.CheckboxText>
          <S.Icon>
            <img alt="Medium" src="./medium.svg" />
          </S.Icon>
          <div>
            <S.Label>Medium</S.Label>
            <S.Description>Followers count</S.Description>
          </div>
        </S.CheckboxText>
      </S.CheckboxField>
    </S.FieldSet>
  </Container.Card>
);

const Notifications = () => (
  <Container.Card heading="Notifications">
    <S.FieldSet>
      <S.Legend>
        Control your notification and auto-follow settings.
      </S.Legend>
      <S.SwitchField description="Commits data and history" label="Email Notification" />
      <S.SwitchField description="Commits data and history" label="Monthly Reports" />
      <S.SwitchField description="Commits data and history" label="Push Notification" />
      <S.SwitchField description="Commits data and history" label="Quarter Reports" />
    </S.FieldSet>
  </Container.Card>
);

export const Theme = () => (
  <Container.Card heading="Theme">
    <S.FieldSet>
      <S.Legend>Select a color scheme for your Velocity app.</S.Legend>
      <S.Theme data-role="shelob">
        <S.Palette />
        <S.RadioField defaultChecked label="Shelob" name="theme" />
      </S.Theme>
      <S.Theme data-role="denethor">
        <S.Palette />
        <S.RadioField label="Denethor" name="theme" />
      </S.Theme>
      <S.Theme data-role="quickbeam">
        <S.Palette />
        <S.RadioField label="Quickbeam" name="theme" />
      </S.Theme>
      <S.Theme data-role="shadowfax">
        <S.Palette />
        <S.RadioField label="Shadowfax" name="theme" />
      </S.Theme>
      <S.Theme data-role="grima">
        <S.Palette />
        <S.RadioField label="Grima" name="theme" />
      </S.Theme>
    </S.FieldSet>
  </Container.Card>
);

export default {
  Personal,
  Integrations,
  Notifications,
  Theme,
};
