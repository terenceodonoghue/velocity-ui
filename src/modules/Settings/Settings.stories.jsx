import React from 'react';
import { storiesOf } from '@storybook/react';
import { Integrations, Notifications, Personal, Theme } from './Settings';

storiesOf('Modules/Settings', module)
  .add('Personal Data', () => <Personal />)
  .add('Integrations', () => <Integrations />)
  .add('Notifications', () => <Notifications />)
  .add('Theme', () => <Theme />);
