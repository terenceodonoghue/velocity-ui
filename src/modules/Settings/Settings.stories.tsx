import React, { FunctionComponent } from 'react';
import * as Settings from './Settings';

export default {
  title: 'Modules/Settings',
};

export const PersonalData: FunctionComponent = () => <Settings.Personal />;

export const Integrations: FunctionComponent = () => <Settings.Integrations />;

export const Notifications: FunctionComponent = () => (
  <Settings.Notifications />
);

export const Theme: FunctionComponent = () => <Settings.Theme />;
