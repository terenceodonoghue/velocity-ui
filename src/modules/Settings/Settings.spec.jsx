import React from 'react';
import { Integrations, Notifications, Personal, Theme } from './Settings';

describe('Settings', () => {
  describe('Personal Data', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(withTheme(<Personal />)).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Integrations', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(withTheme(<Integrations />)).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Notifications', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(withTheme(<Notifications />)).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Theme', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(withTheme(<Theme />)).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
