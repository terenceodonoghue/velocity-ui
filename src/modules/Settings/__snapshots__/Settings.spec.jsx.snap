// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Integrations matches snapshot 1`] = `
.emotion-66 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: #fff;
}

.emotion-0 {
  margin: 0 0 26px 0;
  color: #8798ad;
  font-size: 13px;
  font-weight: 400;
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  text-transform: uppercase;
}

.emotion-64 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin: 0 -11px;
  padding: 0 8px;
}

.emotion-2 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  margin: 0 0 25px 0;
  padding: 0 12px;
  line-height: 1.47;
}

.emotion-12 {
  -webkit-flex: 0 1 365px;
  -ms-flex: 0 1 365px;
  flex: 0 1 365px;
  padding: 13px 11px;
}

.emotion-10 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px #2e5bff;
  border-radius: 5px;
  padding: 19px 21px;
}

.emotion-10::after {
  position: absolute;
  right: -8px;
  top: -8px;
  content: '';
  background-color: #2e5bff;
  background-image: url(./check-white.svg);
  background-position: center;
  background-repeat: no-repeat;
  border-radius: 50%;
  height: 18px;
  width: 18px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  height: 70px;
  margin: 0 12px 0 0;
  width: 70px;
  background-color: rgba(128,151,177,0.1);
}

.emotion-6 {
  margin: 0;
  font-weight: 500;
  line-height: 1.47;
}

.emotion-8 {
  margin: 0;
  color: #8798ad;
  font-weight: 400;
  line-height: 1.47;
}

.emotion-30 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px #e0e7ff;
  border-radius: 5px;
  padding: 19px 21px;
}

<div
  className="emotion-66 emotion-67"
>
  <h3
    className="emotion-0 emotion-1"
  >
    Integrations
  </h3>
  <div
    className="emotion-64 emotion-65"
  >
    <p
      className="emotion-2 emotion-3"
    >
      Manage third-party app integrations.
    </p>
    <label
      className="emotion-12 emotion-13"
    >
      <div
        className="emotion-10 emotion-11"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <img
            alt="InVision"
            src="./invision.svg"
          />
        </div>
        <div>
          <p
            className="emotion-6 emotion-7"
          >
            InVision
          </p>
          <p
            className="emotion-8 emotion-9"
          >
            Boards and prototypes
          </p>
        </div>
      </div>
    </label>
    <label
      className="emotion-12 emotion-13"
    >
      <div
        className="emotion-10 emotion-11"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <img
            alt="GitHub"
            src="./github.svg"
          />
        </div>
        <div>
          <p
            className="emotion-6 emotion-7"
          >
            GitHub
          </p>
          <p
            className="emotion-8 emotion-9"
          >
            Commits data and history
          </p>
        </div>
      </div>
    </label>
    <label
      className="emotion-12 emotion-13"
    >
      <div
        className="emotion-30 emotion-11"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <img
            alt="Slack"
            src="./slack.svg"
          />
        </div>
        <div>
          <p
            className="emotion-6 emotion-7"
          >
            Slack
          </p>
          <p
            className="emotion-8 emotion-9"
          >
            Messages and channels
          </p>
        </div>
      </div>
    </label>
    <label
      className="emotion-12 emotion-13"
    >
      <div
        className="emotion-30 emotion-11"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <img
            alt="Mailchimp"
            src="./mailchimp.svg"
          />
        </div>
        <div>
          <p
            className="emotion-6 emotion-7"
          >
            Mailchimp
          </p>
          <p
            className="emotion-8 emotion-9"
          >
            Subscribers list
          </p>
        </div>
      </div>
    </label>
    <label
      className="emotion-12 emotion-13"
    >
      <div
        className="emotion-30 emotion-11"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <img
            alt="Twitter"
            src="./twitter.svg"
          />
        </div>
        <div>
          <p
            className="emotion-6 emotion-7"
          >
            Twitter
          </p>
          <p
            className="emotion-8 emotion-9"
          >
            Tweets data
          </p>
        </div>
      </div>
    </label>
    <label
      className="emotion-12 emotion-13"
    >
      <div
        className="emotion-30 emotion-11"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <img
            alt="Medium"
            src="./medium.svg"
          />
        </div>
        <div>
          <p
            className="emotion-6 emotion-7"
          >
            Medium
          </p>
          <p
            className="emotion-8 emotion-9"
          >
            Followers count
          </p>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`Settings Notifications matches snapshot 1`] = `
.emotion-42 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: #fff;
}

.emotion-0 {
  margin: 0 0 26px 0;
  color: #8798ad;
  font-size: 13px;
  font-weight: 400;
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  text-transform: uppercase;
}

.emotion-40 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin: 0 -11px;
  padding: 0 8px;
}

.emotion-2 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  margin: 0 0 25px 0;
  padding: 0 12px;
  line-height: 1.47;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-width: 36px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin: 0 87px 0 0;
  padding: 10px 11px;
}

.emotion-4 {
  margin-right: 58px;
  font-weight: 500;
  line-height: 1.47;
}

.emotion-4 p {
  margin: 0;
}

.emotion-4 p:nth-of-type(2) {
  color: #8798ad;
  font-weight: 400;
}

.emotion-6 {
  height: 0;
  margin: 0;
  width: 0;
  opacity: 0;
}

.emotion-8 {
  position: relative;
  border-radius: 10px;
  height: 20px;
  width: 36px;
  background-color: rgba(135,152,173,0.4);
  cursor: pointer;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
}

.emotion-7:disabled + .emotion-8 {
  opacity: 0.4;
}

.emotion-7:checked:enabled + .emotion-8 {
  background-color: #33ac2e;
}

.emotion-8::before {
  position: absolute;
  content: '';
  left: 2px;
  top: 2px;
  border-radius: 50%;
  height: 16px;
  width: 16px;
  background-color: #fff;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
}

.emotion-7:checked + .emotion-8::before {
  -webkit-transform: translateX(16px);
  -ms-transform: translateX(16px);
  transform: translateX(16px);
}

<div
  className="emotion-42 emotion-43"
>
  <h3
    className="emotion-0 emotion-1"
  >
    Notifications
  </h3>
  <div
    className="emotion-40 emotion-41"
  >
    <p
      className="emotion-2 emotion-3"
    >
      Control your notification and auto-follow settings.
    </p>
    <label
      className="emotion-10 emotion-11 emotion-12"
    >
      <div
        className="emotion-4 emotion-5"
      >
        <p>
          Email Notification
        </p>
        <p>
          Commits data and history
        </p>
      </div>
      <input
        className="emotion-6 emotion-7"
        type="checkbox"
      />
      <span
        className="emotion-8 emotion-9"
      />
    </label>
    <label
      className="emotion-10 emotion-11 emotion-12"
    >
      <div
        className="emotion-4 emotion-5"
      >
        <p>
          Monthly Reports
        </p>
        <p>
          Commits data and history
        </p>
      </div>
      <input
        className="emotion-6 emotion-7"
        type="checkbox"
      />
      <span
        className="emotion-8 emotion-9"
      />
    </label>
    <label
      className="emotion-10 emotion-11 emotion-12"
    >
      <div
        className="emotion-4 emotion-5"
      >
        <p>
          Push Notification
        </p>
        <p>
          Commits data and history
        </p>
      </div>
      <input
        className="emotion-6 emotion-7"
        type="checkbox"
      />
      <span
        className="emotion-8 emotion-9"
      />
    </label>
    <label
      className="emotion-10 emotion-11 emotion-12"
    >
      <div
        className="emotion-4 emotion-5"
      >
        <p>
          Quarter Reports
        </p>
        <p>
          Commits data and history
        </p>
      </div>
      <input
        className="emotion-6 emotion-7"
        type="checkbox"
      />
      <span
        className="emotion-8 emotion-9"
      />
    </label>
  </div>
</div>
`;

exports[`Settings Personal Data matches snapshot 1`] = `
.emotion-44 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: #fff;
}

.emotion-0 {
  margin: 0 0 26px 0;
  color: #8798ad;
  font-size: 13px;
  font-weight: 400;
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  text-transform: uppercase;
}

.emotion-39 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin: 0 -11px;
  padding: 0 8px;
}

.emotion-2 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  margin: 0 0 25px 0;
  padding: 0 12px;
  line-height: 1.47;
}

.emotion-7 {
  display: inline-block;
  color: #b0bac9;
  font-size: 12px;
  font-weight: 500;
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  text-transform: uppercase;
  -webkit-flex: 0 1 290px;
  -ms-flex: 0 1 290px;
  flex: 0 1 290px;
  padding: 12px 11px;
}

.emotion-7:first-of-type {
  -webkit-flex: 0 1 100%;
  -ms-flex: 0 1 100%;
  flex: 0 1 100%;
}

.emotion-4 {
  display: block;
  border: solid 1px #e0e7ff;
  border-radius: 5px;
  height: 38px;
  margin: 7px 0;
  max-width: 270px;
  padding: 10px 16px;
  width: 100%;
  background-color: rgba(224,231,255,0.2);
  color: #2e384d;
  font-family: inherit;
  font-size: 15px;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
}

.emotion-4:disabled {
  opacity: 0.3;
}

.emotion-4:focus {
  border: solid 1px #2e5bff;
  outline: 0;
}

.emotion-42 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  padding: 0;
  font-size: 15px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
  border-radius: 4px;
  height: 40px;
  width: 200px;
  background-color: #2e5bff;
  color: #fff;
  margin: 13px 0 18px 8px;
}

.emotion-42:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.emotion-42:focus {
  outline: 0;
}

.emotion-42:enabled:hover {
  background-color: #0036fa;
}

<div
  className="emotion-44 emotion-45"
>
  <h3
    className="emotion-0 emotion-1"
  >
    Personal data
  </h3>
  <div
    className="emotion-39 emotion-40"
  >
    <p
      className="emotion-2 emotion-3"
    >
      Use this page to update your contact information and change your password.
    </p>
    <label
      className="emotion-6 emotion-7 emotion-8"
    >
      Email address
      <input
        className="emotion-4 emotion-5"
        defaultValue="appleseed_jane@mac.com"
        type="text"
      />
    </label>
    <label
      className="emotion-6 emotion-7 emotion-8"
    >
      First name
      <input
        className="emotion-4 emotion-5"
        defaultValue="Jane"
        type="text"
      />
    </label>
    <label
      className="emotion-6 emotion-7 emotion-8"
    >
      Last name
      <input
        className="emotion-4 emotion-5"
        defaultValue="Appleseed"
        type="text"
      />
    </label>
    <label
      className="emotion-6 emotion-7 emotion-8"
    >
      Birth date
      <input
        className="emotion-4 emotion-5"
        defaultValue="1965-05-25"
        type="date"
      />
    </label>
    <label
      className="emotion-6 emotion-7 emotion-8"
    >
      Current password
      <input
        className="emotion-4 emotion-5"
        defaultValue="password"
        type="password"
      />
    </label>
    <label
      className="emotion-6 emotion-7 emotion-8"
    >
      New password
      <input
        className="emotion-4 emotion-5"
        type="password"
      />
    </label>
    <label
      className="emotion-6 emotion-7 emotion-8"
    >
      Confirm
      <input
        className="emotion-4 emotion-5"
        type="password"
      />
    </label>
  </div>
  <button
    className="emotion-41 emotion-42 emotion-43"
  >
    Save Changes
  </button>
</div>
`;

exports[`Settings Theme matches snapshot 1`] = `
.emotion-61 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: #fff;
}

.emotion-0 {
  margin: 0 0 26px 0;
  color: #8798ad;
  font-size: 13px;
  font-weight: 400;
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  text-transform: uppercase;
}

.emotion-59 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin: 0 -11px;
  padding: 0 8px;
}

.emotion-2 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  margin: 0 0 25px 0;
  padding: 0 12px;
  line-height: 1.47;
}

.emotion-13 {
  padding: 8px 11px;
  margin: 0 13px 0 0;
}

.emotion-4 {
  border-radius: 4px;
  height: 65px;
  width: 120px;
  background: linear-gradient(90deg,#2e5bff 33.33%,#e0e7ff 33.33%,#e0e7ff 66.66%,#8097b1 66.66%) no-repeat;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
  padding: 12px 0;
}

.emotion-6 {
  height: 0;
  margin: 0;
  width: 0;
  opacity: 0;
}

.emotion-8 {
  display: inline-block;
  border: solid 1px rgba(128,151,177,0.5);
  border-radius: 50%;
  height: 16px;
  margin: 0 8px 0 0;
  width: 16px;
  background-color: #fff;
}

.emotion-7:disabled + .emotion-8 {
  opacity: 0.4;
}

.emotion-7:checked:enabled + .emotion-8 {
  border: solid 4px #2e5bff;
  background-color: #fff;
}

.emotion-15 {
  border-radius: 4px;
  height: 65px;
  width: 120px;
  background: linear-gradient(90deg,#8c54ff 33.33%,#00c1d4 33.33%,#00c1d4 66.66%,#fad050 66.66%) no-repeat;
}

.emotion-26 {
  border-radius: 4px;
  height: 65px;
  width: 120px;
  background: linear-gradient(90deg,#00a4de 33.33%,#3b7ed5 33.33%,#3b7ed5 66.66%,#00a550 66.66%) no-repeat;
}

.emotion-37 {
  border-radius: 4px;
  height: 65px;
  width: 120px;
  background: linear-gradient(90deg,#232a64 33.33%,#85c800 33.33%,#85c800 66.66%,#616266 66.66%) no-repeat;
}

.emotion-48 {
  border-radius: 4px;
  height: 65px;
  width: 120px;
  background: linear-gradient(90deg,#a728a3 33.33%,#f7e0ff 33.33%,#f7e0ff 66.66%,#7675ac 66.66%) no-repeat;
}

<div
  className="emotion-61 emotion-62"
>
  <h3
    className="emotion-0 emotion-1"
  >
    Theme
  </h3>
  <div
    className="emotion-59 emotion-60"
  >
    <p
      className="emotion-2 emotion-3"
    >
      Select a color scheme for your Velocity app.
    </p>
    <div
      className="emotion-13 emotion-14"
    >
      <div
        className="emotion-4 emotion-5"
      />
      <label
        className="emotion-10 emotion-11 emotion-12"
      >
        <input
          className="emotion-6 emotion-7"
          defaultChecked={true}
          name="theme"
          type="radio"
        />
        <span
          className="emotion-8 emotion-9"
        />
        Shelob
      </label>
    </div>
    <div
      className="emotion-13 emotion-14"
    >
      <div
        className="emotion-15 emotion-5"
      />
      <label
        className="emotion-10 emotion-11 emotion-12"
      >
        <input
          className="emotion-6 emotion-7"
          name="theme"
          type="radio"
        />
        <span
          className="emotion-8 emotion-9"
        />
        Denethor
      </label>
    </div>
    <div
      className="emotion-13 emotion-14"
    >
      <div
        className="emotion-26 emotion-5"
      />
      <label
        className="emotion-10 emotion-11 emotion-12"
      >
        <input
          className="emotion-6 emotion-7"
          name="theme"
          type="radio"
        />
        <span
          className="emotion-8 emotion-9"
        />
        Quickbeam
      </label>
    </div>
    <div
      className="emotion-13 emotion-14"
    >
      <div
        className="emotion-37 emotion-5"
      />
      <label
        className="emotion-10 emotion-11 emotion-12"
      >
        <input
          className="emotion-6 emotion-7"
          name="theme"
          type="radio"
        />
        <span
          className="emotion-8 emotion-9"
        />
        Shadowfax
      </label>
    </div>
    <div
      className="emotion-13 emotion-14"
    >
      <div
        className="emotion-48 emotion-5"
      />
      <label
        className="emotion-10 emotion-11 emotion-12"
      >
        <input
          className="emotion-6 emotion-7"
          name="theme"
          type="radio"
        />
        <span
          className="emotion-8 emotion-9"
        />
        Grima
      </label>
    </div>
  </div>
</div>
`;
